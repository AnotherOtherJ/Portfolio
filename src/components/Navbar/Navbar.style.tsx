import styled from "@emotion/styled";
import { NavLink } from "react-router-dom";

export const Nav = styled.nav`
  display: flex;
  align-items: center;
  height: 70px;
  padding: 0 15px 0 15px;
  gap: 15px;

  > :last-child {
    margin-left: auto;
  }

  svg {
    height: 40px;
    aspect-ratio: 1;
    transition: all 0.5s ease;

    &.logo {
      stroke: ${({ theme: { colors } }) => colors.font};
      :hover {
        stroke: ${({ theme: { colors } }) => colors.accent};
      }
    }

    &.palette {
      user-select: none;
      fill: ${({ theme: { colors } }) => colors.font};
      :hover {
        fill: ${({ theme: { colors } }) => colors.accent};
      }
    }
  }
`;

export const StyledLink = styled(NavLink)`
  color: ${({ theme: { colors } }) => colors.font};
  text-decoration: none;
  user-select: none;

  :hover {
    color: ${({ theme: { colors } }) => colors.accent};
    transition: color 0.5s ease;
  }

  &.active {
    color: ${({ theme: { colors } }) => colors.accent};
    font-weight: 500;
  }
`;
