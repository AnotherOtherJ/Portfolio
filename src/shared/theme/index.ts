export { default } from "./theme";
export type { Themes } from "./theme";
