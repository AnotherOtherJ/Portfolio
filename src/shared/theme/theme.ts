import type { Theme } from "@emotion/react";
export type Themes = Record<string, Theme>;

const dark: Theme = {
  colors: {
    background: "#262626",
    backgroundAcc: "#1F1F1F",
    font: "#FFF",
    primary: "#5B8266",
    accent: "#C8E9A0",
    error: "#BB342F",
  },
};

const light: Theme = {
  colors: {
    background: "#FFF",
    backgroundAcc: "#ddd",
    font: "#FFF",
    primary: "#5B8266",
    accent: "#C8E9A0",
    error: "#BB342F",
  },
};

const theme: Themes = {
  dark,
  light,
};

export default theme;
