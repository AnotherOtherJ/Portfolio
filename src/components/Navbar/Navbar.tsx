import { Nav, StyledLink as NavLink } from "./Navbar.style";
import { ReactComponent as Logo } from "assets/img/logo.svg";
import { ReactComponent as Palette } from "assets/img/palette.svg";
import { Link } from "react-router-dom";
import { useContext, useState } from "react";
import { withTheme } from "@emotion/react";
import { ThemeContext } from "providers";

const Navbar = () => {
  const [showHamburger, setShowHamburger] = useState(false);

  const HandleHamburger = () => {
    setShowHamburger(HamburgerState => !HamburgerState);
  };

  const { dispatch } = useContext(ThemeContext);
  const handleThemeChange = () => {
    dispatch({ type: "change_theme" });
  };

  return (
    <Nav>
      <Link to="/">
        <Logo className="logo" />
      </Link>
      <NavLink to="/">O mnie</NavLink>
      <NavLink to="/skills">Umiejętności</NavLink>
      <NavLink to="/projects">Projekty</NavLink>
      <NavLink to="/experience">Doświadczenie</NavLink>
      <Palette className="palette" onClick={handleThemeChange} />

      <button onClick={HandleHamburger}>Click</button>

      {showHamburger && <div>Lol</div>}
    </Nav>
  );
};

export default withTheme(Navbar);
