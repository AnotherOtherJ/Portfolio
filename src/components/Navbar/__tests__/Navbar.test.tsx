import Navbar from "../Navbar";

import { jsx, ThemeProvider } from "@emotion/react";

import { act, render } from "@testing-library/react";
import theme from "../../../shared/theme/theme";
import { BrowserRouter, Route, Routes } from "react-router-dom";

function AllTheProviders({ children }: any) {
  return (
    <ThemeProvider theme={theme}>
      <BrowserRouter>
        <Routes>
          <Route path="*" element={children} />
        </Routes>
      </BrowserRouter>
    </ThemeProvider>
  );
}

describe("render", () => {
  test("should render", () => {
    const { asFragment } = render(<Navbar />, { wrapper: AllTheProviders });
    expect(asFragment());
  });
});
