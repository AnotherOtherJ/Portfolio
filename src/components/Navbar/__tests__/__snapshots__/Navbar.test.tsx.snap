// Vitest Snapshot v1

exports[`render > should render 1`] = `
{
  "then": [Function],
}
`;
